code/obj/machinery/glass_recycler.dm

```diff
@@ -1,5 +1,34 @@
 #define PLATE_COST 2
 
+/datum/data/glass_product
```

Comment:
This probably doesn't need to be a child of data, but was borrowing ideas from GardenGear.